sort/insertion.py

```diff
@@ -1,7 +1,23 @@
 """
 Insertion sort
 """
+import time
 
 
 def sort(array: list[int]) -> list[int]:
-    return []
+    """Sort by insertion"""
+
+    # For each index of the array
+    for current_index in range(0, len(array)):
+        element_to_sort = array[current_index]
+        # For each element before the element, we compare it to insert it
+        for sub_index in range(current_index - 1, -1, -1):
+            # If the element is inferior to current sub index
+            # and superior to the sub index before,
+            # exception if it equals 0
+            if element_to_sort < array[sub_index] and (sub_index == 0 or
+                                                       element_to_sort >= array[sub_index - 1]):
```

Comment:
Une condition sur plusieurs lignes avec plusieurs and / or est assez difficile à comprendre (sauf au moment où on l'écrit). Dans ce cas, il faut trouver comment découper la complexité un peu.